test/nullable.jl

```diff
@@ -291,5 +295,75 @@ end
 @test Base.promote_op(-, Nullable{Float64}, Nullable{Int}) == Nullable{Float64}
 @test Base.promote_op(-, Nullable{DateTime}, Nullable{DateTime}) == Nullable{Base.Dates.Millisecond}
 
+# getindex
+@test_throws NullException Nullable()[]
+@test_throws NullException Nullable{Int}()[]
+@test Nullable(0)[] === 0
+
+@test_throws BoundsError Nullable()[-1]
+@test_throws BoundsError Nullable()[0]
+@test_throws BoundsError Nullable()[1]
+@test_throws BoundsError Nullable()[2]
+@test_throws BoundsError Nullable()[big(2)^64 + 1]
+@test_throws BoundsError Nullable(0)[-1]
+@test_throws BoundsError Nullable(0)[0]
+@test_throws BoundsError Nullable(0)[2]
+@test Nullable(0)[1] === 0
+
+# filter
+for p in (_ -> true, _ -> false)
+    @test filter(p, Nullable())      |> isnull_oftype(Union{})
```

Comment:
If you apply the `null_safe_op` strategy I described above, you'll be able to do `=== Nullable(true, false)`, since the `value` field will always be equal to `p(x.value)`. That's simpler, and it allows checking that the branch was indeed avoided as expected.

Same below for all basic arithmetic operations on `isbits` types (i.e. everything not involving `BigInt`, `log` or custom/anonymous functions).